cmd/gossamer/main.go

```diff
@@ -224,3 +237,55 @@ func initAction(ctx *cli.Context) error {
 
 	return nil
 }
+
+func buildSpecAction(ctx *cli.Context) error {
+	// set logger to critical, so output only contains genesis data
+	err := ctx.Set("log", "crit")
+	if err != nil {
+		return err
+	}
+	_, err = setupLogger(ctx)
+	if err != nil {
+		return err
+	}
+
+	var bs *dot.BuildSpec
+	if genesis := ctx.String(GenesisFlag.Name); genesis != "" {
+		bspec, e := dot.BuildFromGenesis(genesis)
+		if e != nil {
+			return e
+		}
+		bs = bspec
+	} else {
+		cfg, e := createBuildSpecConfig(ctx)
+		if e != nil {
+			return e
+		}
+		// expand data directory and update node configuration (performed separately
+		// from createDotConfig because dot config should not include expanded path)
+		cfg.Global.BasePath = utils.ExpandDir(cfg.Global.BasePath)
+
+		bspec, e := dot.BuildFromDB(cfg.Global.BasePath)
+		if e != nil {
+			return e
+		}
+		bs = bspec
+	}
+
+	if bs == nil {
+		return fmt.Errorf("error building genesis")
+	}
+
+	res := []byte{}  //nolint
+	if ctx.Bool(RawFlag.Name) {
+		res, err = bs.ToJSONRaw()
+	} else {
+		res, err = bs.ToJSON()
+	}
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%s", res)
```

Comment:
remove

Reply:
This is needed because currently build-spec just dumps the output, so than we can redirect with `>`.  I'll change this when an `--output` flag is added.

Reply:
Added issue #1029